src/ccl_halofit.c

```diff
@@ -0,0 +1,941 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+#include <gsl/gsl_integration.h>
+#include <gsl/gsl_interp.h>
+#include <gsl/gsl_spline.h>
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_roots.h>
+
+#include "ccl.h"
+
+/*
+ * We use the remapping procedure in https://arxiv.org/pdf/1601.07230.pdf
+ * to map w0-wa models onto wa == 0 models.
+ * The functions below implement this procedure.
+*/
+
+// I snuck a private function from backgrounc.c into here. :P
+void compute_chi(double a, ccl_cosmology *cosmo, double * chi, int * stat);
+
+static double zdrag_eh(ccl_parameters *params) {
+  // eqn 4 of Eisenstein & Hu 1998
+  double OMh2 = params->Omega_m * params->h * params->h;
+  double OBh2 = params->Omega_b * params->h * params->h;
+  double b1 = 0.313 * pow(OMh2, -0.419) * (1 + 0.607*pow(OMh2, 0.674));
+  double b2 = 0.238 * pow(OMh2, 0.223);
+  return 1291 * pow(OMh2, 0.251) * (1 + b1*pow(OBh2, b2)) / (1 + 0.659*pow(OMh2, 0.828));
+}
+
+struct hf_model_match_data {
+  double chi_drag;
+  double a;
+  ccl_cosmology *cosmo;
+  int *status;
+};
+
+static ccl_cosmology *create_w0eff_cosmo(double w0eff, ccl_cosmology *cosmo, int *status) {
```

Comment:
Could we just write a couple of sentences describing what these functions are doing?